modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -6361,6 +6830,235 @@ bool GDScriptAnalyzer::class_exists(const StringName &p_class) const {
+	if (return_type == "null") {
+		function_signature += "void";
+	} else {
+		function_signature += return_type;
+	}
+	if (!signature_match) {
+		push_error(vformat(R"("%s" function signature should be "%s" as declared in trait "%s".)", p_source_function->identifier->name, function_signature, p_trait_name), p_target_function);
+	}
+
+	if (p_source_function->is_static && !p_target_function->is_static) {
+#ifdef DEBUG_ENABLED
+		parser->push_warning(p_target_function, GDScriptWarning::UNUSED_STATIC_OVERRIDING_TRAIT);
+#endif
+		p_target_function->is_static = true;
+	}
+	if (p_target_function->rpc_config.get_type() == Variant::NIL) {
+		p_target_function->rpc_config = p_source_function->rpc_config;
+	}
+#ifdef TOOLS_ENABLED
+	copy_over_member_doc_data(p_target_function->doc_data, p_source_function->doc_data);
+#endif
+}
+
+void GDScriptAnalyzer::extend_class(GDScriptParser::ClassNode *p_class, const GDScriptParser::ClassNode *p_trait, const GDScriptParser::Node *p_trait_name_node, const String &p_trait_name) {
+	for (int i = 0; i < p_trait->members.size(); i++) {
+		GDScriptParser::ClassNode::Member trait_member = p_trait->members[i];
+		if (trait_member.type == GDScriptParser::ClassNode::Member::TRAIT) {
+			continue;
+		}
+
+		MethodBind *native_method = ClassDB::get_method(p_class->base_type.native_type, trait_member.get_name());
+		if (native_method) {
+			push_error(vformat(R"("%s" %s from trait "%s" shadows a native method of the using class.)", trait_member.get_name(), trait_member.get_type_name(), p_trait_name), p_trait_name_node);
+			return;
+		}
+		if (has_member_name_conflict_in_native_type(trait_member.get_name(), p_class->base_type.native_type)) {
+			push_error(vformat(R"("%s" %s from trait "%s" shadows a native member of the using class.)", trait_member.get_name(), trait_member.get_type_name(), p_trait_name), p_trait_name_node);
+			return;
+		}
+
+		// Copy over.
+		if (!p_class->has_member(trait_member.get_name())) {
+			if (trait_member.type == GDScriptParser::ClassNode::Member::FUNCTION && trait_member.function->is_bodyless && p_class->type == GDScriptParser::Node::CLASS) {
+				// Unimplemented (Bodyless) Functions.
+				GDScriptParser::FunctionNode *trait_function = trait_member.function;
+				String function_signature = trait_member.get_name() + "(";
+				for (int j = 0; j < trait_function->parameters.size(); j++) {
+					function_signature += trait_function->parameters[j]->get_datatype().to_string_strict();
+					if (j > 0) {
+						function_signature += ",";
+					}
+				}
+				function_signature += ") -> ";
+				const String return_type = trait_function->get_datatype().to_string_strict();
+				if (return_type == "null") {
+					function_signature += "void";
+				} else {
+					function_signature += return_type;
+				}
+				push_error(vformat(R"(Implement function "%s" since class uses "%s" trait.)", function_signature, p_trait_name), p_trait_name_node);
```

Comment:
```suggestion
				push_error(vformat(R"(Class uses trait "%s" but does not implement function "%s".)", p_trait_name, function_signature), p_trait_name_node);
```
Other errors describe the problem with the code rather than giving instructions to solve it, so for consistency I think this should do the same.